tools/c7n_gcp/c7n_gcp/filters/essentialcontacts.py

```diff
@@ -0,0 +1,62 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Essential Contacts suppport for GCP organization
+"""
+from c7n.filters.core import ValueFilter
+from c7n.utils import local_session, type_schema
+from c7n_gcp.resources.resourcemanager import Organization
+import jmespath
+
+
+@Organization.filter_registry.register('essentialcontacts')
+class EssentialContactsFilter(ValueFilter):
+    """Filter Resources based on essential contacts configuration
+
+    .. code-block:: yaml
+
+      - name: org-essential-contacts
+        resource: gcp.organization
+        filters:
+        - type: essentialcontacts
+          org: 99999999
+          category: "ALL"
+    """
+    schema = type_schema('essentialcontacts',
+                        org={'type': 'integer'}, category={'type': 'string'}, required=['org'])
+    required_keys = {}
+    permissions = ("essentialcontacts.contacts.list",)
+    annotation_key = 'c7n:matched-findings'
+
+    def process(self, resources, event=None):
+        self.findings_list = self.get_findings()
+        return [r for r in resources if self.process_resource(r)]
+
+    def get_findings(self):
+        self.findings_by_resource = {}
+        query_params = {
+            'pageSize': 100
+        }
+        session = local_session(self.manager.session_factory)
+        client = session.client("essentialcontacts", "v1", "organizations.contacts")
+
+        findings_paged_list = list(
+            client.execute_paged_query(
+                'list',
+                {'parent': f"organizations/{self.data['org']}", **query_params},
```

Comment:
should this default to the org id of the executing user?

Reply:
@kapilt  Thx for the feedback. 
For org, this filter is similar to gcp filter filter-scc-findings(which also takes org as an input).
There is no environment variable org id that's available in stacklet for gcp, so we will have to derive it for every project this filter runs on.
So, it feels its more flexible to inject it.
But I can change to derive the executing org, if that seems more appropriate.
Please let me know.

Reply:
I can also make it optional, so if the user inputs org, get it from there, otherwise derive the current org of executing user and use that.

Reply:
@kapilt I changed code to now make org optional. If it's not part of input, its derived from the executing gcp project. Please review.
Thx